server/stt/common.py

```diff
@@ -6,27 +6,57 @@
 from superdesk import get_resource_service
 from superdesk.metadata.item import ITEM_TYPE, ITEM_STATE
 from planning.common import WORKFLOW_STATE, POST_STATE, update_post_item
+import re
 
 
 def planning_xml_contains_remove_signal(xml: Element) -> bool:
     """Returns ``True`` if the ``sttinstruct:remove`` signal is included, ``False`` otherwise"""
 
     namespaces = {"iptc": "http://iptc.org/std/nar/2006-10-01/"}
-    if xml.xpath("//iptc:itemMeta/iptc:signal[@qcode='sttinstruct:remove']", namespaces=namespaces):
+    if xml.xpath(
+        "//iptc:itemMeta/iptc:signal[@qcode='sttinstruct:remove']",
+        namespaces=namespaces,
+    ):
         return True
     return False
 
 
 def unpost_or_spike_event_or_planning(item: Dict[str, Any]):
     item_resource = "events" if item.get(ITEM_TYPE) == "event" else "planning"
-    original: Dict[str, Any] = get_resource_service(item_resource).find_one(req=None, _id=item["guid"]) or {}
+    original: Dict[str, Any] = (
+        get_resource_service(item_resource).find_one(req=None, _id=item["guid"]) or {}
+    )
 
     if not original.get("pubstatus") and original.get(ITEM_STATE) in [
         WORKFLOW_STATE.INGESTED,
         WORKFLOW_STATE.DRAFT,
         WORKFLOW_STATE.POSTPONED,
         WORKFLOW_STATE.CANCELLED,
     ]:
-        get_resource_service(item_resource + "_spike").patch(original[config.ID_FIELD], original)
+        get_resource_service(item_resource + "_spike").patch(
+            original[config.ID_FIELD], original
+        )
     elif original.get("pubstatus") != POST_STATE.CANCELLED:
-        update_post_item({"pubstatus": POST_STATE.CANCELLED, "_etag": original["_etag"]}, original)
+        update_post_item(
+            {"pubstatus": POST_STATE.CANCELLED, "_etag": original["_etag"]}, original
+        )
+
+
+def transform_link_from_text(item, fields):
```

Comment:
We already have this code in `superdesk-core` (https://github.com/superdesk/superdesk-core/blob/develop/superdesk/text_utils.py#L193).

This is the code that is currently being used to transform plain text to rich text (for Editor3 fields).
Seems like the bug is there, and should be fixed there

Reply:
I see. Okay, let me check. Thanks, Mark!